ocean_lib/assets/asset_downloader.py

```diff
@@ -16,77 +16,69 @@
 logger = logging.getLogger(__name__)
 
 
-# TODO: fix for v4
-
-
 @enforce_types
-def download_asset_files(
-    service_index: int,
+def download_asset_file(
     asset: Asset,
     consumer_wallet: Wallet,
     destination: str,
-    token_address: str,
     order_tx_id: str,
     data_provider: Type[DataServiceProvider],
-    index: Optional[int] = None,
+    file_index: int,
     userdata: Optional[dict] = None,
 ) -> str:
-    """Download asset data files or result files from a compute job.
+    """Download asset data file or result file from compute job.
 
-    :param service_index: identifier of the service inside the asset DDO, str
     :param asset: Asset instance
     :param consumer_wallet: Wallet instance of the consumer
     :param destination: Path, str
-    :param token_address: hex str the address of the DataToken smart contract
     :param order_tx_id: hex str the transaction hash of the startOrder tx
     :param data_provider: DataServiceProvider class object
-    :param index: Index of the document that is going to be downloaded, int
+    :param file_index: Index of the document that is going to be downloaded, int
+    :param userdata: Dict of additional data from user
     :return: asset folder path, str
     """
-    _files = asset.metadata["main"]["files"]
-    sa = asset.get_service(ServiceTypes.ASSET_ACCESS)
-    service_endpoint = sa.service_endpoint
+    service = asset.get_service(ServiceTypes.ASSET_ACCESS)
+    service_endpoint = service.service_endpoint
     if not service_endpoint:
         logger.error(
             'Consume asset failed, service definition is missing the "serviceEndpoint".'
         )
         raise AssertionError(
             'Consume asset failed, service definition is missing the "serviceEndpoint".'
         )
+    service_id = service.id
+
+    assert isinstance(file_index, int), logger.error("index has to be an integer.")
+    assert file_index >= 0, logger.error("index has to be 0 or a positive integer.")
+
+    service_endpoint += "/api/services/initialize"
+    data_provider.initialize(
```

Comment:
Why don't we store the initialize response?

Reply:
yes, that's the next step. Please use the initialize response in the download inputs.